### PR TITLE
Sync with wwWallet latest master

### PR DESCRIPTION
**Pull request by gkatrakazas**
- Fix redirection loop: extract SessionContext component
- Fix validateDOMNesting and findDOMNode warnings in Login component

### CODE REVIEW

**Comment by emlun** [comment]
I saw that the conflicts were from the `SessionContext` changes, so I took the opportunity to resolve them and pushed that as commit c45552d8366102831ac03b7410fbdeb0d73b78bc on branch `merge-master-into-funke`. I can't easily post that as a meta-PR though, because that would update the `wwwallet-master` branch, and we would have to manually reset that branch in order to not pull the Funke changes into it (I assume you want to keep the `wwwallet-master` branch?). Instead, one can do the merge manually:

```sh
$ git checkout funke
# Show commits added relative to the current state of this PR
$ git log --graph -p funke.. funke/wwwallet-master.. merge-master-into-funke
$ git merge --ff-only merge-master-into-funke
$ git branch -d merge-master-into-funke
$ git push
```

But note that this will also close the pull request as merged, so wait until we're ready to merge :slightly_smiling_face:

**Review by emlun — APPROVED** [review]

**Comment by gkatrakazas** [comment]
> I saw that the conflicts were from the `SessionContext` changes, so I took the opportunity to resolve them and pushed that as commit [c45552d](https://github.com/gunet/funke-wallet-frontend/commit/c45552d8366102831ac03b7410fbdeb0d73b78bc) on branch `merge-master-into-funke`. I can't easily post that as a meta-PR though, because that would update the `wwwallet-master` branch, and we would have to manually reset that branch in order to not pull the Funke changes into it (I assume you want to keep the `wwwallet-master` branch?). Instead, one can do the merge manually:
> 
> ```shell
> $ git checkout funke
> # Show commits added relative to the current state of this PR
> $ git log --graph -p funke.. funke/wwwallet-master.. merge-master-into-funke
> $ git merge --ff-only merge-master-into-funke
> $ git branch -d merge-master-into-funke
> $ git push
> ```
> 
> But note that this will also close the pull request as merged, so wait until we're ready to merge 🙂

We don't necessarily need to update the wwwallet-master. I think you can open a PR with your branch `merge-master-into-funke` to `funke` and just close this one.

**Pull request merged**
